nemo_curator/utils/distributed_utils.py

```diff
@@ -421,34 +421,50 @@ def single_partition_write_with_filename(df, output_file_dir, output_type="jsonl
     else:
         success_ser = pd.Series([empty_partition])
 
-    if empty_partition:
-        filename = df.filename.iloc[0]
-        num_filenames = len(df.filename.unique())
-        if num_filenames > 1:
+    if not empty_partition:
+        filenames = df.filename.unique()
+        filenames = list(filenames.values_host) if is_cudf_type(df) else list(filenames)
+        num_files = len(filenames)
+        if num_files > 1 and output_type != "jsonl":
```

Comment:
We can also do it for `parquet` too , any reason we are skipping that ?

Reply:
The main reason for skipping parquet was the lack of `append` mode in df.to_parquet.
The original functionality of this method is to write out data from a single dataframe to a specific filename/filenames. 
Using `partition_on` will create multiple folders but deviates from the original behavior of writing out files with the same filename as the input.

Reply:
Thanks for clarification. I think we need to support 2 use cases: 

1. Over-write existing file in the directory
2. Append to file in the the directory

Previously we were by default over-writing the file, a behavior i think is a good default because I anticipate folks trying to over-write files instead of wanting to append to existing files and getting confused as a result.  

Suggested Behavior : 


| File Format | Write Mode | Write to filename | Supported |
|-------------|------------|-------------------|-----------|
| jsonl       | Append     | True              |      True     |
| jsonl       | Append     | False             |       True    |
| jsonl       | Overwrite  | True              |      True     |
| jsonl       | Overwrite  | False             |    True       |
| json        | Append     | True              |     True      |
| json        | Append     | False             |      True     |
| json        | Overwrite  | True              |     True      |
| json        | Overwrite  | False             |      True     |
| parquet     | Append     | True              |     False      |
| parquet     | Append     | False             |     False      |
| parquet     | Overwrite  | True              |     True      |
| parquet     | Overwrite  | False             |      True     |